app/src/main/java/com/rockthevote/grommet/ui/registration/address/PersonalInfoFragment.java

```diff
@@ -0,0 +1,84 @@
+package com.rockthevote.grommet.ui.registration.address;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.Nullable;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.rockthevote.grommet.R;
+import com.rockthevote.grommet.data.Injector;
+import com.rockthevote.grommet.databinding.FragmentPersonalInfoBinding;
+import com.rockthevote.grommet.ui.registration.BaseRegistrationFragment;
+import com.rockthevote.grommet.ui.registration.RegistrationData;
+
+import butterknife.ButterKnife;
+import butterknife.OnCheckedChanged;
+import rx.Observable;
+import timber.log.Timber;
+
+
+public class PersonalInfoFragment extends BaseRegistrationFragment {
+
+    private FragmentPersonalInfoBinding binding;
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        binding = FragmentPersonalInfoBinding.inflate(inflater, container, false);
+        return wrapBinding(binding.getRoot(), inflater, container);
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        ButterKnife.bind(this, view);
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        Injector.obtain(getActivity()).inject(this);
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        binding = null;
+    }
+
+    @OnCheckedChanged(R.id.mailing_address_is_different)
+    public void onMailingAddressDifferentChecked(boolean checked) {
+        binding.mailingAddress.setVisibility(checked ? View.VISIBLE : View.GONE);
```

Comment:
was there something about the existing code that prevented using XML data binding? Not saying we should/shouldn't, just curious.

Reply:
Nope, not at all, it's still completely possible. The only reason I went with ViewBinding over DataBinding though is because  there's a lot of two way data relationships because of forms. XML based databinding is great for simply applying a data object to a view, but it's not really convenient when there's a lot of text fields that need to be observed.

The ViewBinding approach also operates pretty similarly to Butterknife. I was hoping to replace some of the butterknife bindings (I did in some places), but don't want to go crazy with it because some of the validation and behavior is tied into butterknife bindings.